docs/aws/audit/snsmonitoring/rules/is_subscription_secure_only.mdx

```diff
@@ -23,6 +23,194 @@ SOC2, HIPAA, ISO27001, AWSWAF
+3. In the SNS dashboard, select "Topics" from the left-hand navigation pane. This will display a list of all the SNS topics available within your AWS account.
+4. For each topic, click on the topic ARN (Amazon Resource Name) to open its details. In the "Subscriptions" section, check the "Protocol" column. If any subscription uses "HTTP" as the delivery protocol, it indicates a misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start, you need to install the AWS CLI on your local machine and configure it with your AWS account credentials. You can do this by running the following commands:
+
+   ```
+   pip install awscli
+   aws configure
+   ```
+
+   You will be prompted to enter your AWS Access Key ID, Secret Access Key, Default region name, and Default output format. These details are available in your AWS Management Console.
+
+2. List all SNS topics: Use the following command to list all the SNS topics in your AWS account:
+
+   ```
+   aws sns list-topics
+   ```
+
+   This command will return a list of all SNS topics along with their ARNs.
+
+3. List all SNS subscriptions: For each SNS topic, list all the subscriptions using the following command:
+
+   ```
+   aws sns list-subscriptions-by-topic --topic-arn <TopicARN>
+   ```
+
+   Replace `<TopicARN>` with the ARN of the SNS topic. This command will return a list of all subscriptions for the specified SNS topic.
+
+4. Check the protocol of each subscription: For each subscription, check the `Protocol` field in the output of the previous command. If the `Protocol` is `http`, then the SNS subscription is using HTTP as the delivery protocol, which is a misconfiguration.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. First, you need to install the AWS SDK for Python (Boto3) in your environment. You can do this by running the command `pip install boto3` in your terminal.
+
+2. Import the necessary modules and create a session using your AWS credentials. Here is a sample script:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2'
+)
+```
+
+3. Now, create a client for SNS and get the list of all SNS subscriptions:
+
+```python
+sns = session.client('sns')
+response = sns.list_subscriptions()
+```
+
+4. Iterate over the list of subscriptions and check the protocol of each subscription. If the protocol is HTTP, print a warning message:
+
+```python
+for subscription in response['Subscriptions']:
+    if subscription['Protocol'] == 'http':
+        print(f"Warning: The SNS subscription {subscription['SubscriptionArn']} uses HTTP as delivery protocol.")
+```
+
+This script will print a warning for each SNS subscription that uses HTTP as the delivery protocol. You can modify it to suit your needs, for example by adding a remediation step or by sending an alert to a monitoring system.
+</Accordion>
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>